gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -75,29 +82,116 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                 File(project.project.buildDir, "kspCaches/$sourceSetName")
     }
 
+    lateinit var artifactName: String
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private val KotlinSourceSet.kspConfigurationName: String
+        get() {
+            return if (name == SourceSet.MAIN_SOURCE_SET_NAME) {
+                KSP_MAIN_CONFIGURATION_NAME
+            } else {
+                "$KSP_MAIN_CONFIGURATION_NAME${name.capitalize(Locale.US)}"
+            }
+        }
+    private fun KotlinSourceSet.kspConfiguration(project: Project): Configuration? {
+        val configName = kspConfigurationName
+
+        return project.configurations.findByName(configName)
+    }
+    @OptIn(ExperimentalStdlibApi::class)
+    private val AndroidSourceSet.kspConfigurationName: String
+        get() {
+            return if (name == SourceSet.MAIN_SOURCE_SET_NAME) {
+                KSP_MAIN_CONFIGURATION_NAME
+            } else {
+                "$KSP_MAIN_CONFIGURATION_NAME${name.capitalize(Locale.US)}"
+            }
+        }
     override fun apply(project: Project) {
         project.extensions.create("ksp", KspExtension::class.java)
-        project.configurations.create(KSP_CONFIGURATION_NAME)
-
+        artifactName = if (project.hasProperty("KSP_ARTIFACT_NAME")) {
+            project.properties.get("KSP_ARTIFACT_NAME") as String
+        } else {
+            DEFAULT_KSP_ARTIFACT_NAME
+        }
+        project.pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+            // kotlin extension has the compilation target that we need to look for to create configurations
+            decorateKotlinExtension(project)
+        }
+        project.pluginManager.withPlugin("com.android.application") {
```

Comment:
You can check for `project.plugins.withType(com.android.build.gradle.api.AndroidBasePlugin) { ... ` in order to detect all plugin types AGP ships.

Reply:
I couldn't use it here because my AGP dependency is compileOnly. That being said, i guess i can reflectively load the class as well.